Can't work in WinUI 3 project

Start from WinUI 3 Preview 3 template project

Add package reference to `Microsoft.Windows.CsWin32`, with `GetActiveWindow` in NativeMethods.txt
Make a call to the method anywhere.

The IDE can correctly analysis all the occurrence and references of the method. However, when building the project, the command line output shows error CS0234: No member 'Windows' in namespace 'Microsoft'.

I think this is https://github.com/dotnet/roslyn/issues/46420 , in another way.

--- a/src/Microsoft.Windows.CsWin32/Generator.cs
+++ b/src/Microsoft.Windows.CsWin32/Generator.cs
@@ -33,8 +33,8 @@ namespace Microsoft.Windows.CsWin32
 
         internal static readonly Dictionary<string, TypeSyntax> BclInteropStructs = new Dictionary<string, TypeSyntax>(StringComparer.Ordinal)
         {
-            { nameof(System.Runtime.InteropServices.ComTypes.FILETIME), ParseTypeName("System.Runtime.InteropServices.ComTypes.FILETIME") },
-            { nameof(Guid), ParseTypeName("System.Guid") },
+            { nameof(System.Runtime.InteropServices.ComTypes.FILETIME), ParseTypeName("global::System.Runtime.InteropServices.ComTypes.FILETIME") },
+            { nameof(Guid), ParseTypeName("global::System.Guid") },
             { "OLD_LARGE_INTEGER", PredefinedType(Token(SyntaxKind.LongKeyword)) },
             { "LARGE_INTEGER", PredefinedType(Token(SyntaxKind.LongKeyword)) },
             { "ULARGE_INTEGER", PredefinedType(Token(SyntaxKind.ULongKeyword)) },

--- a/test/Microsoft.Windows.CsWin32.Tests/GeneratorTests.cs
+++ b/test/Microsoft.Windows.CsWin32.Tests/GeneratorTests.cs
@@ -55,6 +55,10 @@ public class GeneratorTests : IDisposable, IAsyncLifetime
             assemblyName: "test",
             references: metadataReferences,
             options: new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary, allowUnsafe: true));
+
+        // Add a namespace that WinUI projects define to ensure we prefix types with "global::" everywhere.
+        this.compilation = this.compilation.AddSyntaxTrees(
+            CSharpSyntaxTree.ParseText("namespace Microsoft.System { }", this.parseOptions, path: "Microsoft.System.cs"));
     }
 
     public Task DisposeAsync() => Task.CompletedTask;

